### PR TITLE
Add FAQ page and articles

### PR DESCRIPTION
**Pull request by bondtiffanym**

### CODE REVIEW

File: docs/components/CardLinks/index.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import NavCard from './CardLink';
```
Comment (spejamchr) [inline docs/components/CardLinks/index.tsx:2]:
```suggestion
import CardLink from './CardLink';
```

File: docs/faq/about-map.md [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+
+In the mathematical sense, a function is just a mapping from a Set of inputs to a Set of outputs. The output for any given input will always be the same. In the programming world, we call these functions "pure" functions. and they are becoming quite popular.
+
+Pure functions are defined by having no side effects and always returning the same value for the same arguments.This idea is often called "referential integrity", which simply means that if you replace your function call with the value returned from the function, your program doesn't break.
+
+Let's take a moment to explore referential integrity. Given this function:
+
+### add2
+
+        const add2 = (n: number) => n + 2;
```
Comment (spejamchr) [inline docs/faq/about-map.md:24]: In most places we wrap our code blocks in 

````
```typescript 
// code here
```
````

Could we do that in this file too?

````suggestion
```typescript
const add2 = (n: number) => n + 2;
```
````

File: docs/pages/faq.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    'mt-6 space-y-2 border-l-2',
+                    'border-slate-100 dark:border-slate-800',
+                    'lg:mt-4 lg:space-y-4'
+                  )}
+                >
+                  {pages.map(({ slug, frontmatter: { title } }) => (
+                    <li key={slug} className={clsx('relative')}>
+                      <Link href={`/faq/${slug}`}>
+                        <a
+                          href="#"
```
Comment (spejamchr) [inline docs/pages/faq.tsx:51]:
```suggestion
```

Let's remove this. If the anchor tag doesn't have an `href` the `Link` component will fill it in in the html, so the user doesn't need JS to view the page.

File: docs/pages/guide.tsx [hunk 4]
```diff
@@ -7,61 +7,71 @@ import { Page, requireFrontmatterDuringBuild, unsafeMarkdownFromContent } from '
...
+                    'mt-6 space-y-2 border-l-2',
+                    'border-slate-100 dark:border-slate-800',
+                    'lg:mt-4 lg:space-y-4'
+                  )}
+                >
+                  {pages.map(({ slug, frontmatter: { title } }) => (
+                    <li key={slug} className={clsx('relative')}>
+                      <Link href={`/guide/${slug}`}>
+                        <a
+                          href="#"
```
Comment (spejamchr) [inline docs/pages/guide.tsx:51]:
```suggestion
```

Same thing here.

File: docs/components/CardLinks/index.tsx (hunk 1 above)
Comment (bondtiffanym) [reply docs/components/CardLinks/index.tsx:2]: I must have renamed the component at some point and forgot to update, how weird that it didn't throw an error 🤯

File: docs/faq/about-map.md (hunk 2 above)
Comment (bondtiffanym) [reply docs/faq/about-map.md:24]: I was wondering how we had syntax highlighting, thanks!! 😁

**Review by spejamchr — APPROVED** [review]

**Pull request merged**